pkg/datamodeldiagram/datamodelview_test.go

```diff
@@ -18,22 +19,93 @@ func (mock *ClassLabelerMock) LabelClass(className string) string {
 	return args.String(0)
 }
 
-func TestDrawPrimitive(t *testing.T) {
+func assertDraw(t *testing.T,
+	expected, className string,
+	drawFunc func(*DataModelView, map[string]map[string]RelationshipParam)) {
 	clMock := new(ClassLabelerMock)
-	clMock.On("LabelClass", "uuid").Return("test")
+	clMock.On("LabelClass", className).Return("test")
 
 	var stringBuilder strings.Builder
 	v := MakeDataModelView(clMock, nil, &stringBuilder, "title", "project")
-	viewParam := EntityViewParam{
-		EntityColor:  "orchid",
-		EntityHeader: "D",
-		EntityName:   "uuid",
-	}
 	relationshipMap := map[string]map[string]RelationshipParam{}
-	v.DrawPrimitive(viewParam, "INT", relationshipMap)
+	drawFunc(v, relationshipMap)
 	actual := v.StringBuilder.String()
 
-	expected := "class \"uuid\" as _0 << (D,orchid) int >> {\n" + "}\n"
 	assert.EqualValues(t, expected, actual, nil)
 	clMock.AssertExpectations(t)
 }
+
+func TestDrawPrimitive(t *testing.T) {
+	t.Parallel()
+	assertDraw(t,
+		"class \"uuid\" as _0 << (D,orchid) int >> {\n}\n",
+		"uuid",
+		func(v *DataModelView, relationshipMap map[string]map[string]RelationshipParam) {
+			v.DrawPrimitive(
+				EntityViewParam{
+					EntityColor:  "orchid",
+					EntityHeader: "D",
+					EntityName:   "uuid",
+				},
+				"INT",
+				relationshipMap,
+			)
+		},
+	)
+}
+
+func TestDrawTuple(t *testing.T) {
+	t.Parallel()
+	assertDraw(t,
+		`class "aliasName" as _0 << (D,orchid) typeName >> {
++ attr1 : string
+}
+`,
+		"typeName",
+		func(v *DataModelView, relationshipMap map[string]map[string]RelationshipParam) {
+			v.DrawTuple(
+				EntityViewParam{
+					EntityColor:  "orchid",
+					EntityHeader: "D",
+					EntityName:   "typeName",
+					EntityAlias:  "aliasName",
+				},
+				&sysl.Type_Tuple{
+					AttrDefs: map[string]*sysl.Type{
+						"attr1": {
+							Type: &sysl.Type_Primitive_{Primitive: sysl.Type_STRING},
+						},
+					},
+				},
+				relationshipMap,
+			)
+		},
+	)
+
+	assertDraw(t,
+		`class "typeName" as _0 << (D,orchid) >> {
++ attr1 : string
+}
+`,
+		"typeName",
+		func(v *DataModelView, relationshipMap map[string]map[string]RelationshipParam) {
+			v.DrawTuple(
+				EntityViewParam{
+					EntityColor:  "orchid",
+					EntityHeader: "D",
+					EntityName:   "typeName",
+					EntityAlias:  "",
+				},
+				&sysl.Type_Tuple{
+					AttrDefs: map[string]*sysl.Type{
+						"attr1": {
+							Type: &sysl.Type_Primitive_{Primitive: sysl.Type_STRING},
+						},
+					},
+				},
```

Comment:
Could cleanup these braces with some helper functions from syslwrapper. May need to fix the spacing a bit
```suggestion
				syslwrapper.MakeTuple(
				    map[string]*sysl.Type{
				        "attr1": syslwrapper.MakePrimitive("string"),
				    })						
```